apps/worker/src/batch_announcer/ipfs.announcer.ts

```diff
@@ -1,16 +1,334 @@
+      publishStream.on('data', (data) => {
+        buffers.push(data);
+      });
+      publishStream.on('end', () => {
+        resolve(Buffer.concat(buffers));
+      });
+      publishStream.on('error', (err) => {
+        reject(err);
+      });
+    });
+  }
+
+  public async prepareNote(noteContent?: any): Promise<[string, string, string]> {
+    this.logger.debug(`Preparing note`);
+    const tags: ActivityContentTag[] = [];
+    if (noteContent?.content.tag) {
+      noteContent.content.tag.forEach((tag) => {
+        switch (tag.type) {
+          case TagTypeDto.Hashtag:
+            tags.push({ name: tag.name || '' });
+            break;
+          case TagTypeDto.Mention:
+            tags.push({
+              name: tag.name || '',
+              type: 'Mention',
+              id: tag.mentionedId || '',
+            });
+            break;
+          default:
+            throw new Error(`Unsupported tag type ${typeof tag.type}`);
+        }
+      });
+    }
+
+    const attachments: ActivityContentAttachment[] = [];
+    if (noteContent?.content.assets) {
+      noteContent.content.assets.forEach(async (asset: AssetDto) => {
+        switch (asset.type) {
+          case AttachmentTypeDto.LINK: {
+            const link: ActivityContentLink = {
+              type: 'Link',
+              href: asset.href || '',
+              name: asset.name || '',
+            };
+
+            attachments.push(link);
+            break;
+          }
+          case AttachmentTypeDto.IMAGE: {
+            const imageLinks: ActivityContentImageLink[] = [];
+            asset.references?.forEach(async (reference) => {
+              const contentBuffer = await this.ipfsService.getPinned(reference.referenceId);
+              const hashedContent = await calculateDsnpHash(contentBuffer);
+              const image: ActivityContentImageLink = {
+                mediaType: 'image', // TODO
+                hash: [hashedContent],
+                height: reference.height,
+                width: reference.width,
+                type: 'Link',
+                href: await this.formIpfsUrl(reference.referenceId),
+              };
+              imageLinks.push(image);
+            });
+            const imageActivity: ActivityContentImage = {
+              type: 'Image',
+              name: asset.name || '',
+              url: imageLinks,
+            };
+
+            attachments.push(imageActivity);
+            break;
+          }
+          case AttachmentTypeDto.VIDEO: {
+            const videoLinks: ActivityContentVideoLink[] = [];
+            let duration = '';
+            asset.references?.forEach(async (reference) => {
+              const contentBuffer = await this.ipfsService.getPinned(reference.referenceId);
+              const hashedContent = await calculateDsnpHash(contentBuffer);
+              const video: ActivityContentVideoLink = {
+                mediaType: 'video', // TODO
+                hash: [hashedContent],
```

Comment:
Same here , need to get mediaType from references?

Reply:
same as above